src/lib/Form/Type/Content/BaseContentType.php

```diff
@@ -61,8 +67,27 @@ public function buildView(FormView $view, FormInterface $form, array $options)
     public function configureOptions(OptionsResolver $resolver)
     {
         $resolver
-            ->setDefaults(['translation_domain' => 'ibexa_content_forms_content'])
-            ->setRequired(['languageCode', 'mainLanguageCode']);
+            ->setDefaults([
+                'translation_domain' => 'ibexa_content_forms_content',
+                'struct' => null,
```

Comment:
Don't pass default value if you expect developer to always pass this option.
```suggestion
```

Reply:
@Steveb-p isn't the requirement for enabling a certain option using the `setDefaults` method? If we remove `struct` here the option won't get recognized, am I missing something?

Reply:
Well, I have it defined just below:
```php
            ->setRequired(['languageCode', 'mainLanguageCode', 'struct'])
```
and it doesn't seem to catch this option.

Reply:
Okay, nvm, it has to go first

Reply:
Because the order of the operations is incorrect. You cannot `setAllowedTypes` for options that are not yet defined, and `setRequired` method call happens later, after `setAllowedTypes`.

Reverse the order and it will work.

Reply:
I still think this should be defaulted to `null`, otherwise we won't keep BC. The normalizers for standard `ContentEditType`  will normalize `struct` after those options are passed, so after it's `null`. If we won't have this option defaulting to `null` it won't get to normalizer but throw exception because of missing option.

Reply:
It's a class that is not covered by BC promise, as it lies outside of the `Contract` namespace.

Can users/developers outside of Ibexa make use of it? If so, then your comment is valid and it can be left alone. 

However, I'd argue that providing neither `struct` nor any of the deprecated options should be prohibited, as it would cause issues later down the line, when data from this form should be used.

Alternatively if external user/code use has to be considered, then we could indeed make normalizer inject first of the found structs, just like it was before. The reason for making this option required is to make sure that once this form is created, `struct` option is filled.

Otherwise you would need an exception handling for when this option is missing (:warning:)